sdk/cosmos/azure-cosmos-kafka-connect/src/main/java/com/azure/cosmos/kafka/connect/implementation/CosmosConfig.java

```diff
@@ -0,0 +1,237 @@
+        String clientTelemetryEndpoint = this.getString(CLIENT_TELEMETRY_ENDPOINT);
+
+        return new CosmosDiagnosticsConfig(clientTelemetryEnabled, clientTelemetryEndpoint);
+    }
+
+    public static ConfigDef getConfigDef() {
+        ConfigDef configDef = new ConfigDef();
+
+        defineAccountConfig(configDef);
+        defineDiagnosticsConfig(configDef);
+
+        return configDef;
+    }
+
+    private static void defineAccountConfig(ConfigDef result) {
+        final String accountGroupName = "account";
+        int accountGroupOrder = 0;
+
+        // For optional config, need to provide a default value
+        result
+            .define(
+                ACCOUNT_ENDPOINT_CONFIG,
+                ConfigDef.Type.STRING,
+                ConfigDef.NO_DEFAULT_VALUE,
+                new AccountEndpointValidator(),
+                ConfigDef.Importance.HIGH,
+                ACCOUNT_ENDPOINT_CONFIG_DOC,
+                accountGroupName,
+                accountGroupOrder++,
+                ConfigDef.Width.LONG,
+                ACCOUNT_ENDPOINT_CONFIG_DISPLAY
+            )
+            .define(
+                ACCOUNT_KEY_CONFIG,
+                ConfigDef.Type.PASSWORD,
+                ConfigDef.NO_DEFAULT_VALUE,
+                ConfigDef.Importance.HIGH,
+                ACCOUNT_KEY_CONFIG_DOC,
+                accountGroupName,
+                accountGroupOrder++,
+                ConfigDef.Width.LONG,
+                ACCOUNT_KEY_CONFIG_DISPLAY
+            )
+            .define(
+                APPLICATION_NAME,
+                ConfigDef.Type.STRING,
+                Strings.Emtpy,
+                ConfigDef.Importance.MEDIUM,
+                APPLICATION_NAME_DOC,
+                accountGroupName,
+                accountGroupOrder++,
+                ConfigDef.Width.LONG,
+                APPLICATION_NAME_DISPLAY
+            )
+            .define(
+                USE_GATEWAY_MODE,
+                ConfigDef.Type.BOOLEAN,
+                DEFAULT_USE_GATEWAY_MODE,
+                ConfigDef.Importance.LOW,
+                USE_GATEWAY_MODE_DOC,
+                accountGroupName,
+                accountGroupOrder++,
+                ConfigDef.Width.MEDIUM,
+                USE_GATEWAY_MODE_DISPLAY
+            )
+            .define(
+                PREFERRED_REGIONS_LIST,
+                ConfigDef.Type.STRING,
+                Strings.Emtpy,
+                ConfigDef.Importance.HIGH,
+                PREFERRED_REGIONS_LIST_DOC,
+                accountGroupName,
+                accountGroupOrder++,
+                ConfigDef.Width.LONG,
+                PREFERRED_REGIONS_LIST_DISPLAY
+            );
+    }
+
+    private static void defineDiagnosticsConfig(ConfigDef result) {
+        final String diagnosticsGroupName = "diagnostics";
+        int diagnosticsGroupOrder = 0;
+
+        result
+            .define(
+                CLIENT_TELEMETRY_ENABLED,
+                ConfigDef.Type.BOOLEAN,
```

Comment:
We won't need those anymore - will come from account config soon anyway

Reply:
Removed